boot.py

```diff
@@ -1,48 +1,259 @@
+    from .plugin.core.panels import LspUpdatePanelCommand
+    from .plugin.core.panels import LspUpdateServerPanelCommand
+    from .plugin.core.promise import opening_files
+    from .plugin.core.protocol import Response
+    from .plugin.core.protocol import WorkspaceFolder
+    from .plugin.core.registry import LspRecheckSessionsCommand
+    from .plugin.core.registry import LspRestartClientCommand
+    from .plugin.core.registry import windows
+    from .plugin.core.sessions import AbstractPlugin
+    from .plugin.core.sessions import method2attr
+    from .plugin.core.sessions import register_plugin
+    from .plugin.core.sessions import Session
+    from .plugin.core.settings import client_configs
+    from .plugin.core.settings import load_settings
+    from .plugin.core.settings import unload_settings
+    from .plugin.core.transports import kill_all_subprocesses
+    from .plugin.core.types import ClientConfig
+    from .plugin.core.typing import Optional, List, Type, Callable, Dict, Tuple
+    from .plugin.diagnostics import LspHideDiagnosticCommand
+    from .plugin.diagnostics import LspNextDiagnosticCommand
+    from .plugin.diagnostics import LspPreviousDiagnosticCommand
+    from .plugin.documents import DocumentSyncListener
+    from .plugin.documents import TextChangeListener
+    from .plugin.edit import LspApplyDocumentEditCommand
+    from .plugin.execute_command import LspExecuteCommand
+    from .plugin.formatting import LspFormatDocumentCommand
+    from .plugin.formatting import LspFormatDocumentRangeCommand
+    from .plugin.goto import LspSymbolDeclarationCommand
+    from .plugin.goto import LspSymbolDefinitionCommand
+    from .plugin.goto import LspSymbolImplementationCommand
+    from .plugin.goto import LspSymbolTypeDefinitionCommand
+    from .plugin.hover import LspHoverCommand
+    from .plugin.hover import LspRunCommandFromPointCommand
+    from .plugin.panels import LspShowDiagnosticsPanelCommand
+    from .plugin.panels import LspToggleServerPanelCommand
+    from .plugin.references import LspSymbolReferencesCommand
+    from .plugin.rename import LspSymbolRenameCommand
+    from .plugin.save_command import LspSaveCommand
+    from .plugin.selection_range import LspExpandSelectionCommand
+    from .plugin.symbols import LspDocumentSymbolsCommand
+    from .plugin.symbols import LspSelectionAddCommand
+    from .plugin.symbols import LspSelectionClearCommand
+    from .plugin.symbols import LspSelectionSetCommand
+    from .plugin.symbols import LspWorkspaceSymbolsCommand
+    from .plugin.tooling import LspCopyToClipboardFromBase64Command
+    from .plugin.tooling import LspDumpBufferCapabilities
+    from .plugin.tooling import LspDumpWindowConfigs
+    from .plugin.tooling import LspParseVscodePackageJson
+    from .plugin.tooling import LspTroubleshootServerCommand
+
+finally:
+    sys.meta_path.pop(0)
+
+
+def _get_final_subclasses(derived: List[Type], results: List[Type]) -> None:
```

Comment:
This function is going to be needed after all. Because the LSP plugin can be disabled, and then enabled again. In that case, all LSP-* are still loaded so we have to iterate over the final subclasses anyway.